src/tactic/pi_instances.lean

```diff
@@ -14,32 +14,29 @@ Automation for creating instances of mathematical structures for pi types
 
 namespace tactic
 
-open interactive interactive.types lean.parser expr
-open functor has_seq list nat tactic.interactive
+open tactic.interactive
 
-meta def derive_field : tactic unit :=
+/-- Attempt to clear a goal obtained by refining a `pi_instance` goal. -/
+meta def pi_instance_derive_field : tactic unit :=
 do b ← target >>= is_prop,
    field ← get_current_field,
    if b then do
-     field ← get_current_field,
      vs ← introv [] <|> pure [],
      hs ← intros <|> pure [],
-     resetI,
-     x ← get_unused_name,
-     try (() <$ ext1 [rcases_patt.one x] <|> () <$ intro x),
-     x' ← try_core (get_local x),
+     reset_instance_cache,
```

Comment:
Unrelated change here, but looks good.

Reply:
This is #2500. I think something went wrong with the merge.

Reply:
Reverted the rogue commit---sorry